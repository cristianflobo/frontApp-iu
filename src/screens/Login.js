import React  from "react";
import {
  Text,
  StyleSheet,
  View,
  Image,
  TouchableOpacity,
  TextInput,
  ImageBackground,
} from "react-native";
import { fondo, usuario, contrasena } from "../image";
import useLogin from "../hooks/useLogin";
import { login } from "../styles/stylesScreens/login";

export default Login = ({ navigation }) => {
  const { loginUser, dataLogin, setdataLogin } = useLogin(navigation);
  return (
    <View style={styles.conteiner}>
      <ImageBackground source={fondo} resizeMode="cover" style={styles.image}>
        <View style={{ flex: 0.25, alignItems: "center" }}>
          <View style={{ flexDirection: "row" }}>
            <Image style={styles.icon} source={usuario} />
            <TextInput
              style={styles.inputText}
              placeholder="Usuario"
              onChangeText={(newText) =>
                setdataLogin({ ...dataLogin, usuario: newText })
              }
              value={dataLogin.usuario}
            />
          </View>
          <View style={{ flexDirection: "row", marginTop: 15 }}>
            <Image style={styles.icon2} source={contrasena} />
            <TextInput
              secureTextEntry
              style={styles.inputText}
              placeholder="ContraseÃ±a"
              value={dataLogin.password}
              onChangeText={(newText) =>
                setdataLogin({ ...dataLogin, password: newText })
              }
            />
          </View>
        </View>
        <View style={{ flex: 0.3, alignItems: "center" }}>
          <TouchableOpacity style={styles.button} onPress={() => loginUser()}>
            <Text style={styles.testoBoton}>LOGIN</Text>
          </TouchableOpacity>
        </View>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create(login);
