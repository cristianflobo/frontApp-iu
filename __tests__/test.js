import React from "react" // Importante cargar React antes que react-test-renderer
import renderer from "react-test-renderer";
import Login from "../src/screens/Login";


it("renders correctly screen Login", () => {
  const loading  = renderer.create(
    <Login />
  ).toJSON();
  expect(loading).toMatchSnapshot();
});
