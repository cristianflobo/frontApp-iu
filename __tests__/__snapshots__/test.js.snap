// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly screen Login 1`] = `
<View
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../src/image/iuvity.webp",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "flex": 0.25,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/image/usuario.png",
            }
          }
          style={
            {
              "alignItems": "center",
              "height": 30,
              "width": 30,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Usuario"
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#26D07C",
              "height": 40,
              "marginLeft": 9,
              "width": "60%",
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/image/contrasena.png",
            }
          }
          style={
            {
              "alignItems": "center",
              "height": 40,
              "width": 30,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="ContraseÃ±a"
          secureTextEntry={true}
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#26D07C",
              "height": 40,
              "marginLeft": 9,
              "width": "60%",
            }
          }
          value=""
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 0.3,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#26D07C",
            "borderRadius": 50,
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 1,
            "width": 250,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
        >
          LOGIN
        </Text>
      </View>
    </View>
  </View>
</View>
`;
